app/src/main/java/com/vci/vectorcamapp/complete_session/list/presentation/components/CompleteSessionListTile.kt

```diff
@@ -54,15 +55,28 @@ fun CompleteSessionListTile(
     val UPLOAD_ICON_MIN_ALPHA = 0.3f
     val UPLOAD_ICON_MAX_ALPHA = 1f
 
+    val STRONG_SHADOW_ALPHA = 0.4f
+    val WEAK_SHADOW_ALPHA = 0.25f
+
     val session = sessionAndSite.session
     val site = sessionAndSite.site
     val dateFormatter = remember { SimpleDateFormat("MMM dd, yyyy", Locale.getDefault()) }
     val dateTimeFormatter =
         remember { SimpleDateFormat("MMM dd, yyyy 'at' h:mm a", Locale.getDefault()) }
 
+    val progressColor =
+        if (sessionUploadProgress.totalImageCount == 0 || (sessionUploadProgress.uploadedImageCount.toFloat() / sessionUploadProgress.totalImageCount.toFloat()) == 1f) MaterialTheme.colors.primary
+        else if (sessionUploadProgress.isUploading) MaterialTheme.colors.warning
+        else MaterialTheme.colors.error
+
     session.completedAt?.let { completedAt ->
         ActionTile(
             onClick = onClick,
+            shadowColor = if (progressColor == MaterialTheme.colors.primary) progressColor.copy(alpha=STRONG_SHADOW_ALPHA) else progressColor.copy(alpha=WEAK_SHADOW_ALPHA),
+            shadowOffsetX = 0.dp,
+            shadowOffsetY = 0.dp,
+            shadowSpread = MaterialTheme.dimensions.shadowOffsetSmall,
+            shadowBlurRadius = MaterialTheme.dimensions.shadowBlurMedium,
             modifier = modifier,
```

Comment:
Have we tried out making the shadow part of the outer modifier here and removing these custom shadow paramters for the ActionTile?